roles/openshift_health_checker/openshift_checks/diskspace_availability.py

```diff
@@ -0,0 +1,44 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+import json
+
+
+class DiskAvailability(OpenShiftCheck):
+    """Check that recommended disk space is available."""
+
+    name = "disk_availability"
+    tags = ["preflight"]
+
+    recommended_node_diskspace_gb = 15.0
+    recommended_master_diskspace_gb = 40.0
+
+    def run(self, tmp, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        group_names = get_var(task_vars, "group_names", default=[])
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        diskfree = self.to_gigabytes(self.get_disk_space(ansible_mounts, is_containerized))
+
+        recommended_diskspace_gb = self.recommended_master_diskspace_gb
+
+        if "nodes" in group_names:
+            recommended_diskspace_gb = self.recommended_node_diskspace_gb
+
+        if float(diskfree) < recommended_diskspace_gb:
+            return {"failed": True, "msg": "Available disk space for %s (%s) below recommended storage. Minimum required disk space: %s GB" % ("node" if "nodes" in group_names else "master", str(diskfree) + " GB", recommended_diskspace_gb)}
+
+        return {"changed": False}
+
+    @staticmethod
+    def get_disk_space(ansible_mounts, is_containerized):
+        if len(ansible_mounts):
+            for mnt in ansible_mounts:
+                if mnt.get("mount") == "/":
```

Comment:
Technically the requirements are for the filesystem containing `/var` (https://docs.openshift.com/container-platform/3.4/install_config/install/prerequisites.html#hardware). So should check if `/var` is mounted separately before using `/`.